pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -1,8 +1,19 @@
-{ stdenv, fetchurl, ncurses, openssl, perl, python, aspell, gnutls
-, zlib, curl , pkgconfig, libgcrypt, ruby, lua5, tcl, guile
-, pythonPackages, cmake, makeWrapper, libobjc, libiconv
```

Comment:
FWIW, I prefer the first way because it's more precise about the dependencies. If we are concerned about the length of ```all-packages.nix```, I'd prefer to make ```callPackage``` smarter instead of passing the entire ```pythonPackages```.  (I've very quickly tried using ```lib.callPackageWith``` and ```newScope```, but they're causing rebuilds so something is different.)